plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,718 @@
+        iops_rd_max=dict(type='int'),
+        iops_rd_max_length=dict(type='int'),
+        iops_wr=dict(type='int'),
+        iops_wr_max=dict(type='int'),
+        iops_wr_max_length=dict(type='int'),
+        iothread=dict(type='bool'),
+        mbps=dict(type='float'),
+        mbps_max=dict(type='float'),
+        mbps_rd=dict(type='float'),
+        mbps_rd_max=dict(type='float'),
+        mbps_wr=dict(type='float'),
+        mbps_wr_max=dict(type='float'),
+        media=dict(choices=['cdrom', 'disk']),
+        queues=dict(type='int'),
+        replicate=dict(type='bool'),
+        rerror=dict(choices=['ignore', 'report', 'stop']),
+        ro=dict(type='bool'),
+        scsiblock=dict(type='bool'),
+        secs=dict(type='int'),
+        serial=dict(type='str'),
+        shared=dict(type='bool'),
+        snapshot=dict(type='bool'),
+        ssd=dict(type='bool'),
+        trans=dict(choices=['auto', 'lba', 'none']),
+        werror=dict(choices=['enospc', 'ignore', 'report', 'stop']),
+        wwn=dict(type='str'),
+
+        # Disk moving relates parameters
+        bwlimit=dict(type='int'),
+        target_storage=dict(type='str'),
+        target_disk=dict(type='str'),
+        target_vmid=dict(type='int'),
+        delete_moved=dict(type='bool'),
+        timeout=dict(type='int', default='600'),
+
+        # Module related parameters
+        name=dict(type='str'),
+        vmid=dict(type='int'),
+        disk=dict(type='str', required=True),
+        storage=dict(type='str'),
+        size=dict(type='str'),
+        state=dict(choices=['present', 'updated', 'grown', 'detached', 'moved', 'absent'], default='present'),
+        force_replace=dict(type='bool', default=False),
+    )
+
+    module_args.update(disk_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_together=[('api_token_id', 'api_token_secret')],
+        required_one_of=[('name', 'vmid'), ('api_password', 'api_token_id')],
+        required_if=[
+            ('state', 'present', ['storage']),
+            ('state', 'grown', ['size'])
+        ],
+        required_by={
+            'target_disk': 'target_vmid',
+            'mbps_max': 'mbps',
+            'mbps_rd_max': 'mbps_rd',
+            'mbps_wr_max': 'mbps_wr',
+            'bps_max_length': 'mbps_max',
+            'bps_rd_max_length': 'mbps_rd_max',
+            'bps_wr_max_length': 'mbps_wr_max',
+            'iops_max': 'iops',
+            'iops_rd_max': 'iops_rd',
+            'iops_wr_max': 'iops_wr',
+            'iops_max_length': 'iops_max',
+            'iops_rd_max_length': 'iops_rd_max',
+            'iops_wr_max_length': 'iops_wr_max'
+        },
+        supports_check_mode=False,
+        mutually_exclusive=[
+            ('target_vmid', 'target_storage'),
+            ('mbps', 'mbps_rd'),
+            ('mbps', 'mbps_wr'),
+            ('iops', 'iops_rd'),
+            ('iops', 'iops_wr'),
+            ('import_from', 'size')
```

Comment:
For the future: ;-)
```suggestion
            ('import_from', 'size'),
```

Reply:
Is it really necessary?

Reply:
Not "necessary", but it is a good practice.